cms/models.py

```diff
@@ -166,7 +162,7 @@ def texts(self):
 
 
 class Quote(models.Model):
-    partner = models.ForeignKey(Partner)
+    partner = models.ForeignKey(Partner, on_delete=models.CASCADE)
```

Comment:
Are we sure this cascading is safe? Could you explain it to me a little bit?

Reply:
`on_delete` is a required kwarg when defining foreign keys in the newer versions of Django, and it means that when the referred instance is deleted, the instance with this foreign key will also be deleted, so it "cascades" down the chain of relationships.

There are other options which preserve the instance defined [here](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.ForeignKey.on_delete), such as

- `SET` which sets the foreign key field in the database to a predefined value
- `SET_DEFAULT` which sets the foreign key field in the database to the default ForeignKey value
- `SET_NULL` which sets  the foreign key field in the database to `NULL`
- `PROTECT` which prevents deletion of the parent object if it has an existing ForeignKey relation
- `RESTRICT` which is like `PROTECT`, but allows deletion if the object being deleted has another reference to the parent specifying `CASCADE`

Reply:
Sorry I actually meant can you please explain the relationships that it will cascade through? As in, in this case, presumably this means that if a `Quote` is deleted then it will delete the related `Partner`? Is that what it's saying or have I misunderstood?

I'm not quite sure what a "Quote" represents in the partners data model, but it seems unlikely to me that we want a Partner to be deleted when a Quote is deleted doesn't it? I would have thought Partners are pretty important parent items in this data model, and Quotes are relatively short-lived items that come and go. So we would want the act of deleting a Partners to delete all related Quotes perhaps, but we wouldn't want deleting a Quote to delete its parent Partners, that will probably cause chaos.

Or have I misunderstood?

Reply:
It's actually the other way around, so deleting a Partner will delete the related Quote, and any other related objects. 
Though now that you've mentioned it I'm not entirely sure what the desired behavior should be in terms of what we want to keep after deletion.

Reply:
The other models related to Partner are:
- Link
- InsightsTag
- Tag
- Text

Should we cascade delete these as well?

Reply:
Yes I see, you are right, I got the direction wrong so this probably should be safe. Yes, if we're going to cascade on one of the one-to-many relationships from `Partner` then I would have thought we should do all of them.

But I am also wondering, why do you think adding these cascades is necessary for this upgrade? I thought cascades were always supported, and we just hadn't added them. I would imagine this would continue to work just as it was perfectly well without them. I mean, I can see the argument for adding them, but it just doesn't seem necessary specifically to do this upgrade is it?

Because if we do add them then we should probably make sure we're testing deletes thoroughly. It might be simplest just not to add them in this PR and then maybe consider it for a follow-up instead. What do you think?